src/assistants/openai/openai_assistant.rs

```diff
@@ -149,19 +149,7 @@ impl OpenAIAssistant {
     ) -> Result<T> {
         // Instruct the Assistant to answer with the right Json format
         // Output schema is extracted from the type parameter
-        let schema = schema_for!(T);
```

Comment:
Similar code was used several times. I moved it into `utils::get_type_schema` function for better reuse and easier evolution.

Reply:
got it